WCFDataService/Client/System/Data/Services/Client/DataServiceRequest.cs

```diff
@@ -272,6 +273,10 @@ internal long GetQuerySetCount(DataServiceContext context)
 
                 throw;
             }
+            finally
+            {
```

Comment:
Then seems this will cause the response message disposed before emulation as well